File: pythia-0.8/examples/tabulator/tabulator/Exponential.py
#!/usr/bin/env python
#
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#
#                             Michael A.G. Aivazis
#                      California Institute of Technology
#                      (C) 1998-2005  All Rights Reserved
#
# <LicenseText>
#
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#


from pyre.components.Component import Component


class Exponential(Component):


    class Inventory(Component.Inventory):
        
        import pyre.inventory

        a = pyre.inventory.float("a", default=0.0)


    def initialize(self):
        self.a = self.inventory.a
        
        import tabulator._tabulator
        tabulator._tabulator.exponentialSet(self.a)
        return


    def __init__(self):
        Component.__init__(self, "exponential", "functor")
        self.a = 0.0

        import tabulator._tabulator
        self.handle = tabulator._tabulator.exponential()

        return


    def _init(self):
        Component._init(self)
        self.initialize()
        return


# version
__id__ = "$Id: Exponential.py,v 1.1.1.1 2005/03/17 20:03:02 aivazis Exp $"

# End of file 

File: pythia-0.8/examples/tabulator/libtabulator/exponential.f90
! -*- F90 -*-
!
! ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
!
!                             Michael A.G. Aivazis
!                      California Institute of Technology
!                      (C) 1998-2005  All Rights Reserved
!
! <LicenseText>
!
! ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
!

module exponential_parameters
    double precision:: a
end module exponential_parameters

!
subroutine exponential_set(new_a)
    use exponential_parameters

    implicit none
    double precision:: new_a

    a = new_a

    return
end subroutine exponential_set

!
function exponential(x)
    use exponential_parameters

    implicit none

    double precision:: exponential, x

    exponential = dexp(a*x)

end function exponential

! version
! $Id: exponential.f90,v 1.1.1.1 2005/03/17 20:03:02 aivazis Exp $

! End of file 
